### DISCUSSION TITLE
Imposed strain in bar elements

### DISCUSSION THREAD

**Discussion by jose-ortolano** [BriefFiniteElementNet/BriefFiniteElement.Net#185] [Q&A]
Hi!

I'm doing some tests with Brief Finite Element.NET. I am trying to figure out if it is possible somehow to impose a strain in some bars. My main goal is simulating the effect of loads such as temperature or prestressing.

Any information or help will be appreciated. Thank you in advance!

Kind Regards,

**Comment by epsi1on** [comment]
hi, there is currently no straight way to take into account thermal expansion or imposed strain.
both easy to implement, continue in #186 .

**Comment by epsi1on** [comment]
i think you can emulate such effect with superposition.
if and element have imposed strain of say `ε`, two opposite nodal load with amount of `P=ε*E*A` applied to both ends will emulate the imposed strain.
so if you apply two opposite nodal loads with that magnitude then target element will try to expand and whole structure will behave as what you need. the only different is when you want to get internal force of the target element, you need to negate  `P` from the value you get from `BarElement.GetInternalForce()` like this pseudo code: 
`axialForce = BarElement.GetInternalForce().Fx - P`

**Comment by jose-ortolano** [comment]
Hi!

Thank you four your quick answer! The solution you mention is not valid for what I want to do. The imposition of an action of this type must be done in terms of imposed deformation and not through force.

If a force is imposed on a node where more bars meet, the effect of the force will affect each bar depending on the stiffness of each one. On the other hand, I need to apply a deformation with a certain value on a specific bar, independently of the rest of the bars. It is a subtle difference, however, on certain systems the differences can be important.

Thanks again for the reply.

Cheers!

**Reply by epsi1on** [reply]
the good thing about linear FEA is it's simplicity. i think if you use equivalent nodal load as my above post (formula is edited) you will get exact same nodal displacement with imposed strain. then only thing you should worry is internal force of the strain imposed element...

**Reply by epsi1on** [reply]
any success?

**Comment by jose-ortolano** [comment]
Hi! 

I'm sorry for not having answered before. The solution of adding forces at bar ends did not work for my case.

However, I managed to emulate the prestressing effect of the cables (I was analyzing a guyed tower)  using imposed displacements on one of the ends of the cables (which were anchored in the ground). 

Since the analysis included cables (they cannot withstand compression loads), I also had to implement a manual incremental calculation to determine when to eliminate cables subjected to compression forces. Although it is not 100% accurate, it helped me for the preliminary analysis I was doing.

Thank you very much for your help.

Kind regards,

![GuyedTower](https://github.com/BriefFiniteElementNet/BriefFiniteElement.Net/assets/141654076/01c0ec30-dbf9-461a-b770-95ecb1b1a7b4)

**Reply by epsi1on** [reply]
ok, i suggest you to keep eye on #186 . will close this discussion for now.

**Reply by epsi1on** [reply]
Hi, i've added a new load named `ImposedStrainLoad`, which can apply imposed strain on any element.
Could you please have a check and see if it works correct?

**Reply by jose-ortolano** [reply]
Hi! Thank you for having implemented! 

I will try it and I will check if it works correctly. I will confirm you if it works properly as soon as I can.

Cheers,
